Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -182,6 +182,14 @@ class BrowserViewController: UIViewController {
     private func updateToolbarStateForTraitCollection(newCollection: UITraitCollection) {
         let showToolbar = shouldShowFooterForTraitCollection(newCollection)
         let showTopTabs = shouldShowTopTabsForTraitCollection(newCollection)
+        
+        if let mvc = menuViewController where showToolbar != (toolbar != nil) {
+            // Hide the menu, and then re-open it so that the menu is always the correct one for the given traits
+            mvc.dismissViewControllerAnimated(true, completion: nil)
+            dispatch_async(dispatch_get_main_queue()) {
+                self.tabToolbarDidPressMenu(self.navigationToolbar, button: self.navigationToolbar.menuButton)
```

Comment:
Edit: it turns out that the menu is reöpened too soon when it's in the completion handler due to the way split screen works. It could probably be dispatched _within_ the completion handler, although that seems superfluous. Anywhere later in the method produces distorted menus too.